src/ExperimentalFunds.mo

```diff
@@ -0,0 +1,54 @@
+/// Funds and units
+///
+/// Provides imperative operations for observing funds, transferring funds and
+/// observing refunds of various units.
+/// The two currently support units are `#cycle` and `#icpt`, measuring
+/// computational cost and Internet Computer tokens, respectively.
+
+/// This low-level API is experimental and likely to change, including the
+/// addition of other units and alternative syntactic support for
+/// manipulating funds.
+///
+/// Note: since unit `#cycle` is used to measure computation, the value of
+/// `balance(#cycle)` generally decreases from one call to the next.
+
+import Prim "mo:prim";
+module {
+
+  /// Units for funds: `{#cycle; #icpt}`.
+  public type Unit = Prim.Unit;
+
+  /// Returns the actor's current balance of unit `u` as `amount`.
+  public let balance : (u : Unit) -> (amount : Nat64) = Prim.fundsBalance;
+
+  /// Given `u`, returns the currently available `amount` of unit `u`.
+  /// The amount available is the amount received in the current message,
+  /// minus the cumulative amount `accept`ed by this message.
```

Comment:
```suggestion
  /// The amount available is the amount received in the current call,
  /// minus the cumulative amount `accept`ed by this call.
```
(This spans multiple messages, i.e. also the callbacks, associated with the current call, right?)

Reply:
Yes. Provided that's what available does in the System API (that's what I understood from the spec)